README.md

```diff
@@ -1 +1,67 @@
-# Bigcommerce Sample App: Hello World in Ruby and Sinatra
+# Bigcommerce Sample App: Ruby
+This is a Sinatra small application that implements the OAuth callback flow for [Bigcommerce Single Click Apps][single_click_apps]
+and uses the [Bigcommerce API][api_client] to pull a list of products on a Bigcommerce store. For information on how to develop apps
+for Bigcommerce stores, see our [Developer Portal][devportal].
+
+We hope this sample app gives you a good starting point for building your next killer app! What follows are steps specific
+to running and installing this sample appication.
+
+### Registering the app with Bigcommerce
+* Create a trial store on [Bigcommerce](https://www.bigcommerce.com/)
+* Go to the [Developer Portal][devportal] and log in.
+* Go to "My Apps"
+* Click the button "Create an app", enter a name for the new app, and then click "Create"
+* You don't have to fill out all the details for your app right away, but you do need
+to provide some core details in section 4 (Technical). Note that if you are just getting
+started, you can use `localhost` for your hostname, but ultimately you'll need to host your
+app on the public Internet.
+  * _Auth Callback URL_: `https://<app hostname>/auth/bigcommerce/callback`
+  * _Load Callback URL_: `https://<app hostname>/load`
+  * _Uninstall Callback URL_: `https://<app hostname>/uninstall`
+* Enable the _Products - Modify_ scope under _OAuth scopes_. This sample app uses that scope, but if you need others,
+you'll need to specify them here and then modify the `scopes` method in `hello.rb` to match.
+* Click `Save & Close` on the top right of the dialog.
+* You'll now see your app in a list in the _My Apps_ section of Develper Portal. Hover over it and click
+_View Client ID_. You'll need these values in the next step.
+
+### Getting started
+* Clone this repo: `git clone git@github.com:bigcommerce/hello-world-app-ruby-sinatra.git`
+* Change to the repo directory: `cd hello-world-app-ruby-sinatra`
+* Install gems: `bundle install --path vendor/bundle`
+* Copy `.env-example` to `.env`
+* Edit `.env`:
+  * Set `BC_CLIENT_ID` and `BC_CLIENT_SECRET` to the values obtained from Developer Portal.
+  * Set `APP_URL` to `https://<app hostname>`
+  * Set `SESSION_SECRET` to a long random string
```

Comment:
true :(

date and md5 should not be used in a production it's predictable, something generated using `dev/urandom` based on openssl lib, or ruby's `SecureRandom` since the secret is used to sign/validate cookies